javascript/index.js

```diff
@@ -168,21 +170,21 @@ function encodeHeader(precision, thirdDim, thirdDimPrecision) {
 function encodeUnsignedNumber(val) {
     // Uses variable integer encoding to encode an unsigned integer. Returns the encoded string.
     let res = '';
-    let bigIntVal = BigInt(val);
+    let bigIntVal = Num(val);
```

Comment:
I'm not happy with the name `bigIntVal` since it is no longer necessarily a BigInt. can you rename to `integerVal` (or some other name if you find a suitable one) please?